tests/test_source_sampling.py

```diff
@@ -69,7 +71,9 @@ def test_analog_multiple_hex():
     m.src = IMeshTag(2, float)
     m.src[:] = np.ones(shape=(8,2))
     m.mesh.save("sampling_mesh.h5m")
-    sampler = Sampler("sampling_mesh.h5m", "src", np.array([0, 0.5, 1]), False)
+    filename = "sampling_mesh.h5m"
+    tag_names = {"src_tag_name": "src"}
```

Comment:
I'm surprised that this test passes given the missing `bias_tag_name` key - see comment above.

Reply:
This test passes because the `bias_tag_name = tag_names["bias_tag_name"]` happens in C++ part.
In C++, the `map::operator[]` searches the data structure for a value corresponding to the given key, and returns a reference to it. If it can't find one it transparently creates a default constructed element for it, rather than report an error. (In python, it will report an error if the key doesn't exist.)
I have added an input check in the code and add a test function to make sure that it will report an error with corresponding information of which input parameter is missing.